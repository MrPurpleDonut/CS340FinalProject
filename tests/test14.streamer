funk num sum(list<num> nums)
{
  num result = 0;
  for (num n : nums)
  {
    result = result + n;
  }
  return result;
}

funk void printResult(num result, text start)
{
  print start + "\n" + result;
}

printResult(sum({86, 7, 5, 309}), "Result");
