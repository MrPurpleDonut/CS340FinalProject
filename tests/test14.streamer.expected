Program parsed correctly!
---------------------------
--------Printing AST-------
---------------------------
Program
    Function List
        Function declaration: sum
            Type: num
            Parameter List
                Parameter: nums
                    Type: list
                        Type: num
            Statement List
                Declaration statement: result
                    Type: num
                    Assignment statement: result
                        Number: 0.0
                For Statement: n
                    Type: num
                    Identifier: nums
                    Statement List
                        Assignment statement: result
                            Binary operator '+'
                                Identifier: result
                                Identifier: n
                Return Statement
                    Identifier: result
        Function declaration: printResult
            Type: void
            Parameter List
                Parameter: result
                    Type: num
                Parameter: start
                    Type: text
            Statement List
                Print Statement
                    Binary operator '+'
                        Binary operator '+'
                            Identifier: start
                            Quoted string: "\n"
                        Identifier: result
    Statement List
        Statement as Expression
            Function call: printResult
                Expression List
                    Function call: sum
                        Expression List
                            Expression List
                                Number: 86.0
                                Number: 7.0
                                Number: 5.0
                                Number: 309.0
                            Quoted string: "Result"
